backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropLowering.kt

```diff
@@ -1193,6 +1213,21 @@ private class InteropTransformer(val context: Context, override val irFile: IrFi
                         putValueArgument(3, jobPointer)
                     }
                 }
+                IntrinsicType.CREATE_CLEANER -> {
+                    val irCallableReference = expression.getValueArgument(1)
+                    if (irCallableReference == null || !irCallableReference.isNonCapturingLambda()) {
+                        context.reportCompilationError(
```

Comment:
@homuroll FYI: one more compilation error reported from backend, please make sure it doesn't get lost when merging this PR with your one.

Reply:
Ok, thanks for notifying.